src/attr/_make.py

```diff
@@ -194,6 +203,21 @@ def wrap(cl):
             cl = _add_hash(cl)
         if init is True:
             cl = _add_init(cl)
+        if slots:
+            cl_dict = dict(cl.__dict__)
+            cl_dict['__slots__'] = tuple(ca_list)
+            for ca_name in ca_list:
+                # It might not actually be in there, f.e. if using 'these'.
+                cl_dict.pop(ca_name, None)
```

Comment:
Yeah that’s interesting.   But yeah, we can’t just add everything to slots as we want so it’s okay.